flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/filemerging/FileMergingSnapshotManagerBase.java

```diff
@@ -619,6 +636,7 @@ public void restoreStateHandles(
                                         fileHandle.getStartPos(),
                                         fileHandle.getStateSize(),
                                         subtaskKey);
+                        spaceStat.onLogicalFileCreate(logicalFile.getLength());
```

Comment:
This should also be protected by `managedByFileMergingManager` right? I'd suggest adding an attribute to the `PhysicalFile` to mark whether this file is managed by the `managedByFileMergingManager`, and reading this each time the logical file related stats update.